docs/github_projects_guidelines.md

```diff
@@ -44,68 +44,59 @@ working with projects hosted on GitHub.
 ## Branching model 🌲
 
 <p align="center">
-  <img src="../images/gitflow-atlassian.svg" alt="gitflow" width=600 ><br><br>
-    <i>GitFlow branching diagram, source: <a href="https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow">Atlassian</a></i><br>
+  <img src="../images/trunk-based-development.png" alt="trunk" width=800 ><br><br>
+    <i>Trunk-based development branching model, source: <a href="https://statusneo.com/trunk-based-development/">StatusNeo</a></i><br>
 
 </p>
 
-The IRNAS's Git branching model is based on the [GitFlow] branching model with
-some small differences.
+The IRNAS's Git branching model is based on the [Trunk-based development]
+branching model with some minor differences.
 
 ### Key concepts
 
-Each repository needs to have two long-lived branches, `master` and `dev`.
+#### Trunk aka. `main`
 
-- `master` branch should always contain production-ready code, which should be
-  committed to it in form of a GitHub Pull Request. Nothing should be committed
-  to it directly (exceptions can be made for GitHub CI workflows, which can
-  otherwise misbehave).
-- `dev` branch contains the latest development changes for the next release.
-  Commits to it can be done to it directly or in form of GitHub Pull requests.
+Each repository has a single long-lived, default, branch named `main`.
 
-Besides these two branches, various short-lived branches exist in this workflow.
-They are classified into three categories: feature, release, and hotfix.
+There are two rules that should be followed when adding new code to the `main`:
 
-- feature branches are used to develop new features. They always branch from and
-  back into the `dev` branch, never in `master`. They are named as `feature/*`,
-  for example, the names `feature/new-board` and `feature/refactor` are both
-  valid feature branch names.
-- release branches are used for the preparation of a new release. They allow for
-  last-minute changes. They are always created from the `dev` branch and merged
-  into the `master` branch. They are named as `release/<version>`, where
-  `<version>` indicates the version of the next release.
-- hotfix branches are used to address unplanned changes to the release in
-  production. The essence is that the work of team members can continue, while
-  another person is preparing a quick production fix. Hotfix branches must
-  always branch from the `master` branch and merge back into it.
+- It should **never break the build** (whatever _build_ means for a specific
+  repository). Bugs should be caught by the CI or by code reviews.
+- `main` should **always be release ready**, even if there is work in progress.
+  Functionalities that might need some more development time to become useful
+  should not be included in the build until they are ready.
```

Comment:
Maybe add some examples of how to achieve this.

For example, by either not merging a feature branch until it is ready. Not including unfinished files in CMake, ...

Reply:
Added.